src/lib_json/json_reader.cpp

```diff
@@ -1096,8 +1090,8 @@ bool OurReader::parse(const char* beginDoc,
 }
 
 bool OurReader::readValue() {
-  if (stackDepth_ >= features_.stackLimit_) throwRuntimeError("Exceeded stackLimit in readValue().");
-  ++stackDepth_;
+  //  To preserve the old behaviour we cast size_t to int.
+  if (static_cast<int>(nodes_.size()) > features_.stackLimit_) throwRuntimeError("Exceeded stackLimit in readValue().");
```

Comment:
stack<>.size() complexity is constant: stack is deque (http://www.cplusplus.com/reference/stack/stack/) and deque<>.size complexity - http://www.cplusplus.com/reference/deque/deque/size/

About OurReader. I have change it because of the following comment to the class:
// exact copy of Reader, renamed to OurReader
and 
// complete copy of Read impl, for OurReader

If you allow some small differences I would rollback my changes to OurReader.